importer/BDCS/CS.hs

```diff
@@ -118,6 +143,57 @@ store repo content =
         repoWriteArchiveToMtree repo archive mtree Nothing True noCancellable
         repoWriteMtree repo mtree noCancellable
 
+-- Given an open ostree repo and a checksum, read an object from the content store.
+-- The checksums stored in the mddb can be either dirmeta objects, which will contain
+-- the metadata for a directory (mode, xattrs), or a file object, which will contain the file
+-- metadata plus contents.
+-- On error (such as if the object does not exist) a String exception is thrown.
+load :: (IsRepo a, MonadError String m, MonadIO m) => a -> T.Text -> m Object
+load repo checksum =
+    ifM (repoHasObject repo ObjectTypeDirMeta checksum noCancellable)
+        (DirMeta <$> loadDir)
+        (ifM (repoHasObject repo ObjectTypeFile checksum noCancellable)
+             (FileObject <$> loadFile)
+             (throwError $ "No such object " ++ show checksum))
```

Comment:
You could use condM here instead to avoid the nesting:
```
condM [(repoHasObject ..., DirMeta <$> loadDir),
       (repoHasObject ..., FileObject <$> loadFile),
       (otherwiseM, throwError ...)]
```